uhyve-interface/src/elf.rs

```diff
@@ -0,0 +1,47 @@
+//! Utility to place the uhyve interface version in the elf header of a hermit application.
+
+/// Defines the hermit entry version in the note section.
+///
+/// This macro must be used in a module that is guaranteed to be linked.
+/// See <https://github.com/rust-lang/rust/issues/99721>.
+#[macro_export]
+macro_rules! define_uhyve_interface_version {
+	() => {
+		#[used]
+		#[link_section = ".note.hermit.uhyve-interface-version"]
+		static INTERFACE_VERSION: $crate::elf::Note = $crate::elf::Note::uhyveif_version();
+	};
+}
+
+/// Note type for specifying the uhyve interface version in an elf header.
+pub const NT_UHYVE_INTERFACE_VERSION: u32 = 0x5b00;
+
+/// A elf note header entry containing the used Uhyve interface version as little-endian 32-bit value.
+#[repr(C)]
+pub struct Note {
+	header: Nhdr32,
+	name: [u8; 8],
+	data: [u8; 4],
+}
+
+impl Note {
+	pub const fn uhyveif_version() -> Self {
+		Self {
+			header: Nhdr32 {
+				n_namesz: 8,
+				n_descsz: 4,
```

Comment:
Did you test these? They _should_ be fine, but in my experience these numbers are quite fragile.

Reply:
No, but they are in line with the documentation. Eight bytes for ` b"UHYVEIF\0"` and four for the u32 `UHYVE_INTERFACE_VERSION`.

Reply:
I know, but that's not enough and that's why I ask. When I tested my numbers, when I introduced them, the obvious numbers did not work, and I had to tweak them to get them working.

You can confirm if they are read correctly with some binutils command. Which one specifically, I can't remember from the top of my head.

Reply:
```sh
readelf target/x86_64/debug/libhermit.a -n
#[...]
Displaying notes found in: .note.hermit.entry-version
  Owner                Data size        Description
  HERMIT               0x00000001       Unknown note type: (0x00005a00)
   description data: 02

Displaying notes found in: .note.hermit.uhyve-interface-version
  Owner                Data size        Description
  UHYVEIF              0x00000004       Unknown note type: (0x00005b00)
   description data: 00 00 00 01
#[...]
```
I guess this is correct this way

Reply:
Looks good 👍